test/Kestrel.FunctionalTests/BadHttpRequestTests.cs

```diff
@@ -169,6 +172,86 @@ public Task BadRequestIfHostHeaderDoesNotMatchRequestTarget(string requestTarget
             Assert.Equal(LogLevel.Information, maxLogLevel);
         }
 
+        [Fact]
+        public async Task OverlongHeaderIsRejected()
+        {
+            using (var server = CreateServer())
+            {
+                using (var client = server.CreateConnection())
+                {
+                    await client.Send(
+                        "GET / HTTP/1.1",
+                        "Host:",
+                        "Connection: close");
+
+                    var payload = new string('x', 102400);
+                    client.Socket.Send(Encoding.ASCII.GetBytes(String.Format("Range: {0}\r\n\r\n", payload)));
+
+                    await client.ReceiveStartsWith("HTTP/1.1 431");
+                }
+            }
+        }
+
+        [Fact]
+        public async Task HostHeaderAcceptedWithPeriod()
+        {
+            using (var server = CreateServer())
+            {
+                using (var client = server.CreateConnection())
+                {
+                    await client.Send(
+                        "GET / HTTP/1.1",
+                        "Host: .",
+                        "Connection: close\r\n\r\n");
+
+                    await client.ReceiveStartsWith("HTTP/1.1 200");
+                }
+            }
+        }
+
+        [Fact]
+        public async Task UnprintableCharactersAreFiltered()
+        {
+            using (var server = CreateServer())
+            {
+                using (var client = server.CreateConnection())
+                {
+                    await client.Send(
+                        "GET /\x0D\0x0ALocation:http://www.contoso.com/ HTTP/1.1",
+                        "Host:\r\n\r\n");
```

Comment:
Isn't this the same as
```
"GET /",
"Location:http://www.contoso.com/ HTTP/1.1",
"Host:\r\n\r\n"
```
And it gets rejected because the request line is missing `HTTP/1.1`? This doesn't seem to be testing unprintable character filtering.

Reply:
I'm adding `"GET /\r\nLocation:http://www.contoso.com/ HTTP/1.1\r\n"` to `HttpParsingData.RequestLineInvalidData`, then this will get tested by `BadHttpRequestTests.TestInvalidRequestLines`.

Reply:
Actually I'll leave it in its own test.